src/Cech_complex/include/gudhi/Cech_complex_blocker.h

```diff
@@ -30,33 +31,111 @@ namespace cech_complex {
  * `simplex_vertex_range(Simplex_handle sh)`and `assign_filtration(Simplex_handle sh, Filtration_value filt)` methods.
  *
- * \tparam Chech_complex is required by the blocker.
+ * \tparam Cech_complex is required by the blocker.
+ *
+ * \tparam Kernel CGAL kernel: either Epick_d or Epeck_d.
  */
-template <typename SimplicialComplexForCech, typename Cech_complex>
+template <typename SimplicialComplexForCech, typename Cech_complex, typename Kernel>
 class Cech_blocker {
+
+ public:
+
+  using Point_d = typename Kernel::Point_d;
+  // Numeric type of coordinates in the kernel
+  using FT = typename Kernel::FT;
+  // Sphere is a pair of point and squared radius.
+  using Sphere = typename std::pair<Point_d, FT>;
+
  private:
-  using Point_cloud = typename Cech_complex::Point_cloud;
 
   using Simplex_handle = typename SimplicialComplexForCech::Simplex_handle;
   using Filtration_value = typename SimplicialComplexForCech::Filtration_value;
 
+  template<class PointIterator>
+  Sphere get_sphere(PointIterator begin, PointIterator end) const {
+    Point_d c = kernel_.construct_circumcenter_d_object()(begin, end);
+    FT r = kernel_.squared_distance_d_object()(c, *begin);
+    return std::make_pair(std::move(c), std::move(r));
+  }
+
  public:
+
   /** \internal \brief Čech complex blocker operator() - the oracle - assigns the filtration value from the simplex
    * radius and returns if the simplex expansion must be blocked.
    *  \param[in] sh The Simplex_handle.
    *  \return true if the simplex radius is greater than the Cech_complex max_radius*/
   bool operator()(Simplex_handle sh) {
-    Point_cloud points;
-    for (auto vertex : sc_ptr_->simplex_vertex_range(sh)) {
-      points.push_back(cc_ptr_->get_point(vertex));
+    using Point_cloud =  std::vector<Point_d>;
+    CGAL::NT_converter<FT, Filtration_value> cast_to_fv;
+    Filtration_value radius = 0;
+    bool is_min_enclos_ball = false;
+
+    // for each face of simplex sh, test outsider point is indeed inside enclosing ball, if yes, take it and exit loop, otherwise, new sphere is circumsphere of all vertices
+    for (auto face : sc_ptr_->boundary_simplex_range(sh)) {
+        // Find which vertex of sh is missing in face. We rely on the fact that simplex_vertex_range is sorted.
+        auto longlist = sc_ptr_->simplex_vertex_range(sh);
+        auto shortlist = sc_ptr_->simplex_vertex_range(face);
+
+        auto longiter = std::begin(longlist);
+        auto shortiter = std::begin(shortlist);
+        auto enditer = std::end(shortlist);
+        while(shortiter != enditer && *longiter == *shortiter) { ++longiter; ++shortiter; }
+        auto extra = *longiter; // Vertex_handle
+
+        Sphere sph;
+        auto k = sc_ptr_->key(face);
+        if(k != sc_ptr_->null_key()) {
+            sph = cc_ptr_->get_cache().at(k);
+        }
+        else {
+            Point_cloud face_points;
```

Comment:
(for later) Reuse the vector (and `points` later), or at least `reserve()` or use a small_vector, to avoid wasting time in reallocations.

Reply:
cf. #652